### PR TITLE
Implement issue #329 (callback with arguments for input-ports)

### PR DESCRIPTION
**Pull request by folkertvanheusden**
Introduces 'callback_w_args' callback-parameter for input-ports.
This implements issue #329

### CODE REVIEW

File: examples/callback_w_args/callback_w_args.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import time
+
+rtmidi = mido.Backend('mido.backends.rtmidi')
+
+def myfunc(msg, q):
+    print(f'Got message: {msg}')
+    q.put(msg)
+
+q = queue.Queue()
+input_ = rtmidi.open_input('testing', virtual=True, callback_w_args=(myfunc, q))
```
Comment (smiszym) [inline examples/callback_w_args/callback_w_args.py:12]: There's no need to implement anything in mido to achieve this. The standard way would be to use [`functools.partial`](https://docs.python.org/3.9/library/functools.html#functools.partial) to create a new callable that has some parameters set to something. (if I understand correctly what you want to achieve). Like this:

```py
input_ = rtmidi.open_input('testing', virtual=True, callback=partial(myfunc, q=q))
```

Comment (folkertvanheusden) [reply examples/callback_w_args/callback_w_args.py:12]: Thank you! I did not know about "partial".

**Comment by folkertvanheusden** [comment]
Change is not required.

File: examples/callback_w_args/callback_w_args.py (hunk 1 above)
Comment (smiszym) [reply examples/callback_w_args/callback_w_args.py:12]: Good question. Indeed you could use partial with Thread too. I guess args is there just for convenience as starting a Thread with some arguments is quite common.

**Pull request closed without merging**